replica-node/src/replica_util.rs

```diff
@@ -0,0 +1,268 @@
+use {
+    log::*,
+    rand::{seq::SliceRandom, thread_rng, Rng},
+    solana_gossip::{
+        cluster_info::{ClusterInfo, Node},
+        contact_info::ContactInfo,
+        gossip_service::GossipService,
+    },
+    solana_runtime::snapshot_utils,
+    solana_sdk::{
+        clock::Slot,
+        hash::Hash,
+        pubkey::Pubkey,
+        signature::{Keypair, Signer},
+    },
+    std::{
+        collections::HashSet,
+        net::{SocketAddr, UdpSocket},
+        path::Path,
+        process::exit,
+        sync::{atomic::AtomicBool, Arc},
+        thread::sleep,
+        time::{Duration, Instant},
+    },
+};
+
+pub fn get_cluster_shred_version(entrypoints: &[SocketAddr]) -> Option<u16> {
```

Comment:
Is this identical to the method in `validator/src/main.rs`?
If so, can we dedupe by moving this to the solana-gossip crate?

Reply:
This implementation is exactly the same

Reply:
I will deduce it.

Reply:
Actually I will keep it -- the code might be removed altogether when we change to remove the dependency on using gossip to discover initial snapshots. I talked to @sakridge on it to have it addressed in another PR.

Reply:
Actually I will keep it -- the code might be removed altogether when we change to remove the dependency on using gossip to discover initial snapshots. I talked to @sakridge on it to have it addressed in another PR.